main.py

```diff
@@ -975,6 +998,62 @@ def on_interface_deleted(self, event):
     def handle_interface_deleted(self, event):
         """Update the topology based on a Port Delete event."""
         self.handle_interface_down(event)
+        interface = event.content['interface']
+        usage = self.get_intf_usage(interface)
+        if usage:
+            log.info(f"Interface {interface.id} could not be safely removed."
+                     f" Reason: {usage}")
+        else:
+            self._delete_interface(interface)
+
+    def get_intf_usage(self, interface: Interface) -> Optional[str]:
+        """Determines how an interface is used explained in a string,
+        returns None if unused."""
+        if interface.is_enabled() or interface.is_active():
```

Comment:
Interface active attr currently maps down to the protocol state. I have an impression that it might be too restrictive here. Can you double check in the OF1.3 spec if there's any requirement that will ensure that it's always going to send OFPT_PORT_STATUS which would deactivate it? In a example in practice on OvS I'm seeing that it's doing it, but let's confirm with the spec.

Reply:
The spec says that for port status message: `As ports are added, modified, and removed from the datapath, the controller needs to be informed with the OFPT_PORT_STATUS message.`
Also under port specification: `All port state bits are read-only and cannot be changed by the controller. When the port flags are changed, the switch sends an OFPT_PORT_STATUS message to notify the controller of the change.`

One thing though is that if the port is logically existing, after a deletion and kytos restart, the port will be appear again (it does not get rediscovered otherwise). The new port will be enabled.

Reply:
> It sound like it always sends OFPT_PORT_STATUS when the status of a port changes.

Yes, that's correct according to the spec, but from what I followed, there's no guarantee about which PortState it'll be, and here it's assuming it's always going to be not active, in practice so far with the examples it was always down though, but that's just one OvS implementation, it'd be great if we found more details otherwise, we'll need to make this a bit more flexible, at the end of the day, it's also validating whether or not that interface has a link, so already having that check is great, and an active interface being deleted without a link and without any flows shouldn't be an issue I believe.  

> One thing though is that if the port is logically existing, after a deletion and kytos restart, the port will be appear again (it does not get rediscovered otherwise). The new port will be enabled.

If a new `OFPPR_ADD` and or port description is sent with a recreated interface, it should be rediscovered, and then enabled if using the flag `-E` it's expected to be auto enabled, otherwise it should be enabled with the endpoint.

Reply:
Let's leave the `interface.active()` check in place. Since we've been already treating an interface deletion as deactivation, and there was an deactivation still being forced there, so it's OK assuming we didn't have any issues with that approach https://github.com/kytos-ng/topology/blob/3cfb5ca8523167253cae2eb714bef54d16103b7b/main.py#L990

I'll close this thread.